validator/src/commands/run/execute.rs

```diff
@@ -455,19 +436,20 @@ pub fn execute(
         values_t!(matches, "account_shrink_path", String)
             .map(|shrink_paths| shrink_paths.into_iter().map(PathBuf::from).collect())
             .ok();
-    let account_shrink_paths = account_shrink_paths.as_ref().map(|paths| {
-        create_and_canonicalize_directories(paths).unwrap_or_else(|err| {
-            eprintln!("Unable to access account shrink path: {err}");
-            exit(1);
+    let account_shrink_paths = account_shrink_paths
+        .as_ref()
+        .map(|paths| {
+            create_and_canonicalize_directories(paths)
+                .map_err(|err| format!("unable to access account shrink path: {err}"))
         })
-    });
+        .transpose()?;
```

Comment:
I used `.transpose()` a decent bit in this PR. As a sanity check, you can plug something like this into Rust playground to prove to yourself that it is correct:
```rust
    #[test]
    fn test_transpose() {
        let four: Option<Result<u8>> = Some(Ok(4));
        let none: Option<Result<u8>> = None;
        assert_eq!(four.transpose().unwrap(), Some(4));
        assert_eq!(none.transpose().unwrap(), None);
    }
```